test/integration/restart_task_test.rb

```diff
@@ -117,8 +162,45 @@ def test_restart_not_existing_namespace
       namespace: "walrus",
       logger: logger
     )
-    refute restart.perform(["web"])
-    assert_logs_match("Namespace `walrus` not found in context `minikube`. Aborting the task.")
+    assert_restart_failure(restart.perform(["web"]))
+    assert_logs_match_all([
+      "Result: FAILURE",
+      "Namespace `walrus` not found in context `minikube`"
+    ],
+      in_order: true)
+  end
+
+  def test_restart_failure
+    success = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      deployment = fixtures["web.yml.erb"]["Deployment"].first
+      deployment["spec"]["progressDeadlineSeconds"] = 12
```

Comment:
Why 12?

Reply:
Trial and error... just trying to pick a number that allows a bit of time for the expected logs/events to be reported but doesn't make the test too slow.